docs/source/concepts/local-ledger.rst

```diff
@@ -0,0 +1,452 @@
+      `act`:sub:`1` and `act`:sub:`2` in `X` must be action-ordered.
+      For each such pair, determine the actions' ordering in `G` and add an edge to `G'` from the earlier action's transaction to the later action's transaction.
+   #. `reduce`:sub:`X`\ `(G)` is the transitive closure of `G'`.
+
+Topological sort and reduction link causality graphs `G` to the ledgers `L` from the DAML Ledger Model.
+Topological sort transforms a causality graph `G` into a sequence of transactions `topo(G)`; extending them with the requesters gives a sequence of commits, i.e., a ledger in the DAML Ledger Model.
+Conversely, a sequence of commits `L` yields a causality graph `G`:sub:`L` by taking the transactions as vertices and adding an edge from `tx1` to `tx2` whenever `tx1`\ 's commit precedes `tx2`\ 's commit in the sequence.
+
+There are now two consistency definitions:
+
+* :ref:`Ledger Consistency <da-model-ledger-consistency>` according to DAML Ledger Model
+
+* :ref:`Consistency of causality graph <def-consistency-causality-graph>`
+
+Fortunately, the two definitions are equivalent:
+If `G` is a consistent causality graph, then `topo(G)` is ledger consistent.
+Conversely, if the sequence of commits `L` is ledger consistent, `G`:sub:`L` is a consistent causality graph, and so is the reduction `reduce(G`:sub:`L`\ `)`.
+
+.. _local-ledger-structure:
+   
+Local ledgers
+*************
+
+As explained in the DAML Ledger Model, parties see only a :ref:`projection <da-model-projections>` of the shared ledger for privacy reasons.
+Like consistency, projection extends to causality graphs as follows.
+
+Definition »Causal consistency for a party«
+  A causality graph `G` is **consistent for a party** `P` (`P`-consistent) if `G` is consistent on all the actions that `P` is an informee of.
+
+The notions of `X`\ -minimality and `X`\ -reduction extend to parties accordingly.
+
+For example, the :ref:`counteroffer causality graph without the edge tx2 -> tx4 <causality-counteroffer-Iou-minimal>` is consistent for the Bank because the Bank is an informee of exactly the highlighted actions.
+It is also minimal Bank-consistent and the Bank-reduction of the :ref:`original counteroffer causality graph <causality-graph-couteroffer-split>`.
+  
+Definition »Projection of a consistent causality graph«
+  The **projection** `proj`:sub:`P`\ `(G)` of a consistent causality graph `G` to a party `P` is the `P`\ -reduction of the following causality graph `G'`:
+
+  * The vertices of `G'` are the vertices of `G` projected to `P`, excluding empty projections.
+
+  * There is an edge between two vertices `v`:sub:`1` and `v`:sub:`2` in `G'` if there is an edge from the `G`\ -vertex corresponding to `v`:sub:`1` to the `G`\ -vertex corresponding to `v`:sub:`2`.
+
+For the :ref:`Counteroffer causality graph <causality-graph-couteroffer-split>`, the projections to Alice, the Bank, and the painter are as follows.
```

Comment:
This is the "split counteroffer", do we consistently (what a pun!) distinguish between the split and the "normal" one in the doc?

Reply:
No, we didn't. I've added a lot more `split` to the text to make it uniform.